src/app/context/appState.tsx

```diff
@@ -157,7 +162,23 @@ function actionHandler(draft: Draft<AppState>, action: AppStateAction) {
       draft.isLoading = false;
       return;
     }
+    case 'SAVE_AS_IUM': {
+      saveAsIum(draft, action.payload?.filename, action.payload?.spaceIndent);
+      return;
+    }
```

Comment:
You should not have side effects in a reducer.

Actions should only update the state.